mne/preprocessing/annotate_nan.py

```diff
@@ -0,0 +1,47 @@
+# Author: David Julien <david.julien@ifsttar.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+import warnings
+
+from ..annotations import Annotations
+from ..utils import _mask_to_onsets_offsets
+
+
+def annotate_nan(raw):
+    """Detect segments with NaN and return a new Annotations instance.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to find segments with NaN values.
+
+    Returns
+    -------
+    annot : instance of Annotations
+        Updated annotations for raw data.
+    """
+    annot = raw.annotations.copy()
+    data, times = raw.get_data(return_times=True)
+    sampling_duration = 1 / raw.info['sfreq']
+
+    nans = np.any(np.isnan(data), axis=0)
+    starts, stops = _mask_to_onsets_offsets(nans)
+
+    if len(starts) > 0:
```

Comment:
I would switch this conditional to be `if len(starts) == 0: warn() return` that way the lines below this can be unidented.